src/components/Portal/Portal.js

```diff
@@ -109,6 +109,7 @@ class Portal extends Component {
         this.overlay.setAttribute('tabIndex', '-1');
         this.overlay.classList.add(`${namespace}__overlay`);
         rootNode.appendChild(this.overlay);
+        this.overlay.addEventListener('mousedown', this.handleOverlayClick);
```

Comment:
I wonder if we should only apply this event listener if `onOverlayClick` has been defined by the consumer, to avoid it always executing an empty method. We may be able to prevent this by setting the `defaultProp` to `null` or `undefined` and wrapping this line in a conditional.

Reply:
For clean up, we'll also need to ensure the DOM event is removed when the component unmounts.